### PR TITLE
Add subjects mapping support in nats chart

### PR DESCRIPTION
**Pull request by OpenGuidou**
According to the doc here: https://docs.nats.io/running-a-nats-service/configuration/configuring_subject_mapping

### CODE REVIEW

File: helm/charts/nats/values.yaml [hunk 1]
```diff
@@ -196,6 +196,35 @@ nats:
...
 
+  # Subject mapping
+  #
+  #   You can find more info at: https://docs.nats.io/running-a-nats-service/configuration/configuring_subject_mapping
+  #
+  # subjectsMapping:
+  #   - source: foo
+  #     destinations:
+  #       - subject: "bar"
+  #   - source: foo.cluster.scoped,
```
Comment (skreborn) [inline helm/charts/nats/values.yaml:207]: Are the commas at the ends of lines supposed to be there?

Comment (OpenGuidou) [reply helm/charts/nats/values.yaml:207]: > Are the commas at the ends of lines supposed to be there?

You're right it's an issue in the doc only

**Comment by OpenGuidou** [comment]
@skreborn Is it ok now ?

**Comment by skreborn** [comment]
> @skreborn Is it ok now ?

Oh, me, I'm all good, nothing to complain about, but I unfortunately have no power over moving this along.

**Comment by wallyqs** [comment]
toPrettyJson would escape html so if there are wildcards it would become html entities, maybe toRawJson could work

**Comment by caleblloyd** [comment]
ah good point thanks @wallyqs - yea that would look quite a bit uglier when read out but at least we would not have to maintain the schema in `values.yaml`.  updated my code snippet to `toRawJson`

**Comment by OpenGuidou** [comment]
It seems to linearize the config and the server doesn't like that:

```
###################################
    #                                 #
    # NATS subjects mapping           #
    #                                 #
    ###################################
    mappings: " { foo: bar foo.cluster.scoped: [ { destination: bar.cluster.scoped, weight: 70%, cluster: us-west-1 }, { destination: foobar.cluster.scoped, weight: 30%, cluster: us-east-1 } ] } "
```
nats-server: /etc/nats-config/nats.conf:19:1: interface conversion: interface {} is string, not map[string]interface {}

**Comment by OpenGuidou** [comment]
Thanks for the review.
I updated it based on your comments, and tested it successfully on my side.

File: helm/charts/nats/templates/configmap.yaml [hunk 2]
```diff
@@ -105,6 +105,14 @@ data:
       {{- end }}
     }
     {{- end }}
+    {{- if .Values.nats.subjectsMapping }}
```
Comment (caleblloyd) [inline helm/charts/nats/templates/configmap.yaml:108]:
```suggestion
    {{- if .Values.nats.mappings }}
```

File: helm/charts/nats/templates/configmap.yaml [hunk 3]
```diff
@@ -105,6 +105,14 @@ data:
...
+    ###################################
+    #                                 #
+    # NATS subjects mapping           #
```
Comment (caleblloyd) [inline helm/charts/nats/templates/configmap.yaml:111]:
```suggestion
    # Mappings                        #
```

File: helm/charts/nats/templates/configmap.yaml [hunk 4]
```diff
@@ -105,6 +105,14 @@ data:
...
+    #                                 #
+    ###################################
+    mappings: {{ toRawJson .Values.nats.subjectsMapping }}
```
Comment (caleblloyd) [inline helm/charts/nats/templates/configmap.yaml:114]:
```suggestion
    mappings: {{ toRawJson .Values.nats.mappings }}
```

**Review by caleblloyd — COMMENTED** [review]
looks good, going to rename `.Values.nats.subjectMappings` to `.Values.nats.mappings` to match the NATS config key exactly

**Comment by caleblloyd** [comment]
Need to run a couple tests then I will add approval if all working

**Review by caleblloyd — APPROVED** [review]
LGTM.  Thanks for the contribution!

**Pull request merged**